### PR TITLE
feat(design system): add hedgehog illustration components

### PR DESCRIPTION
**Pull request by kappa90**
## Problem

We currently don't have ready-made hedgehogs in the app and the design system, but we have them in Figma.

## Changes
- Created components for all the hedgehogs...
- ...and a new beautiful library of illustrations in Storybook

Screenshot 2022-09-02 at 15 56 28

## How did you test this code?
- Visually tested in Storybook

### CODE REVIEW

**Review by benjackwhite — COMMENTED** [review]
I love this ❤️!

Just checking, this doesn't instantly up our bundle size right? As in the images aren't bundled... right?

**Comment by kappa90** [comment]
> I love this ❤️!
> 
> Just checking, this doesn't instantly up our bundle size right? As in the images aren't bundled... right?

I am not sure, and that's why I requested a review, I knew this question would be asked 😆

File: frontend/src/lib/components/hedgehogs.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import professorHog from 'public/hedgehog/professor-hog.png'
+import supportHeroHog from 'public/hedgehog/support-hero-hog.png'
+import xRayHog2 from 'public/hedgehog/x-ray-hogs-02.png'
+import laptopHog3 from 'public/hedgehog/laptop-hog-03.png'
+
+export const SurprisedHog = (props: ImgHTMLAttributes<HTMLImageElement>): JSX.Element => {
+    return <img src={surprisedHog} width={200} height={200} {...props} />
+}
+export const XRayHog = (props: ImgHTMLAttributes<HTMLImageElement>): JSX.Element => {
+    return <img src={xRayHog} width={200} height={200} {...props} />
```
Comment (pauldambra) [inline frontend/src/lib/components/hedgehogs.tsx:29]: the idea `width={200} height={200}` is represented many times in the same file

I wonder if it's worth having an unexported `BaseHog` that sets the width and height to make it clear that individual hogs don't get to set width and height

**Review by pauldambra — APPROVED** [review]
I get 25.5MB of resources with an empty cache on this branch and on master

One nit but otherwise :shipit:

**Review by lottiecoxon — APPROVED** [review]
this looks great ! thank you so much @kappa90

**Review by mariusandra — COMMENTED** [review]
Great hogs! I don't see any bundler issues with the way they're included.

However, how hard would it be to have them as `.svg` files instead of `.png` files? As they're now, and taking retina screens into account (1 screen pixel equals 4 device pixels), we can basically show them as 100x100 px images. That might not be enough.

**Comment by lottiecoxon** [comment]
@mariusandra Figma has the option to export as an SVG but I think the quality would be bad unless you export at a larger size than we have now and scale down? (unsure if this would take up more storage and affect load times) - alternatively I could embark on creating hedgehogs on Adobe illustrator (which is what I do when animating them) which would allow for higher quality hedgehogs at a smaller size - issue with this would be that it would take a long time to,  a) catch up on all hedgehogs we have now and b) would take up a lot more time drawing them up with that software. (Procreate is just a hell of a lot quicker).

**Comment by mariusandra** [comment]
SVGs are normally smaller than PNGs and can render at any resolution without losing image quality (they might just look weird if zoomed in/out too much).

However:

> "Procreate is a raster graphics editor app"

Ah, got it. SVGs are not possible in this case.

If we're going with raster images, then the hedgehogs should be a bit larger. If the intent is to show them as 200x200px images (measured by "web pixels" - the posthog app's sidebar is 250px for example), then the png files should actually be 400x400 pixels. Otherwise they'll look blurry or fuzzy on all "retina" screens, meaning basically any mac.

**Comment by mariusandra** [comment]
One more tip when working with PNGs. Install and use a program called [ImageOptim](https://imageoptim.com/mac). Before committing any images to github, drag them into this window. It does some magic and usually chops off ~15% of the image's size without losing any quality.

image

**Comment by kappa90** [comment]
@lottiecoxon if you can export 400x400 images on Figma, I'll replace and optimize them for storybook!

**Comment by lottiecoxon** [comment]
@kappa90 where shall I put/send the compressed 400x400 versions?

**Comment by kappa90** [comment]
> @kappa90 where shall I put/send the compressed 400x400 versions?

Can you maybe update them on Figma?

**Comment by lottiecoxon** [comment]
Done - in the design system now!

**Comment by kappa90** [comment]
I updated all the hedgehogs, updated a couple ones that were already in use, and fixed a bug in `utilities.scss` (`max-w-full` and `max-w-full` were wrong).

**Pull request merged**